src/chapters/az-cli/walkthrough_az-cli.rst

```diff
@@ -0,0 +1,537 @@
+.. _walkthrough_az-cli:
+
+=======================================================
+Walkthrough: Provisioning Resources Using the Azure CLI
+=======================================================
+
+In this walkthrough we will practice using the ``az CLI`` to provision the same resources that we spun up using the web portal GUI. As you work through these commands try to picture the manual steps you performed on the web portal and use your knowledge of the ``az CLI`` patterns to translate those steps into CLI commands.
+
+Although you can simply read through this guide it is important that you actively attempt each command on your machine. The more practice you get the more comfortable you will be with using it.
+
+For each resource we provision we will use a methodical approach:
+
+- **Exploration**: issue a help command to learn about the the resource
+- **Planning**: consider our tasks and what Commands, Sub-Groups and Arguments are needed to accomplish them
+- **Provisioning**: put our Planning into action and provision the resource
+- **Configuring**: any extra steps that use the new resource to complete our tasks
+
+This is a process you can carry with you as you continue to work with the ``az CLI`` and other tooling. A good idea is to start a set of notes for easy access to common tasks you are faced with. These notes will prove invaluable when you begin composing commands together into automation scripts. 
+
+.. note::
+
+    As a cross-platform tool all of the base commands will work the same on any OS or shell. In some cases we will utilize shell-specific syntax to accomplish our tasks. Where there are syntactical differences the commands for both Linux/BASH and Windows/PowerShell will be presented.
+
+Setup
+=====
+
+Before we begin you have to log in with your Azure subscription. This command will open your browser to enter your credentials. After logging in you can close the tab and return to your terminal.
+
+.. sourcecode:: powershell
+
+    > az login
```

Comment:
do students know this is a powershell command they are entering?

Reply:
i had a note above that unless the command is shell specific then it is cross-platform. added another sentence for clarity

```
.. note::

    As a cross-platform tool all of the base commands will work the same on any OS or shell. In some cases we will utilize shell-specific syntax to accomplish our tasks. Where there are syntactical differences the commands for both Linux/BASH and Windows/PowerShell will be presented. Otherwise you can freely use the commands in either shell.
```